### PR TITLE
[WIP] Use local built CSS on master documentation

### PR DESCRIPTION
**Pull request by ylemkimon**
* Copy CSS and fonts before `start`, `build`, and `publish`

### CODE REVIEW

**Comment by ylemkimon** [comment]
The concern is commiting CSS and fonts may bloat the repo.

File: docs/supported.md [hunk 1]
```diff
@@ -2,8 +2,9 @@
 id: supported
 title: Supported Functions
 ---
+<link rel="stylesheet" href="/KaTeX/static/katex.min.css"/><!-- must use absolute path -->
```
Comment (ylemkimon) [inline docs/supported.md]: - [ ] Change to use `baseUrl` from siteConfig

File: website/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "scripts": {
-    "prestart": "cp ../docs/cli.md.template ../docs/cli.md && node ../cli.js markdown-help >> ../docs/cli.md",
+    "prestart": "cp ../docs/cli.md.template ../docs/cli.md && node ../cli.js markdown-help >> ../docs/cli.md && cp -r ../dist/katex.min.css ../dist/fonts static/static",
```
Comment (ylemkimon) [inline website/package.json]: Instead of copying files, we can use build artifacts from Circle CI build. However this needs a CORS proxy.

**Comment by ronkok** [comment]
@ylemkimon I fully approve of the content in the `supported.md` file. I also agree completely with the objectives of the other changes in this PR and they all seem to work smoothly emprically.

But regarding build scripts, I'm probably not qualified to pass judgment. Perhaps @kevinbarabash  or @edemaine could weigh in.

**Comment by edemaine** [comment]
I guess the main concern here is repo bloat in the `gh-pages` branch.

I'd like to reraise the question of whether we should have a separate repo for the documentation/website publication (not the source files).  We could move the website to `http://katex.github.io/` which would be served from a repository called `katex.github.io` within the KaTeX project.  (See https://help.github.com/articles/user-organization-and-project-pages/#user-and-organization-pages-sites)  Then we wouldn't have to worry about repo size, and we would be free to put all the versions of documentation there (which really wants to have the CSS/JS/fonts copied).

@kevinbarabash Thoughts?  Should we discuss this in a separate issue?

**Comment by ylemkimon** [comment]
I think we can discuss in #1509.

**Comment by ylemkimon** [comment]
@ronkok This PR only changes two lines of `supported.md`, so I think it'll be easy to resolve merge conflicts.

File: website/package.json (hunk 2 above)
Comment (kevinbarabash) [reply website/package.json]: Using build artifacts from Circle CI sounds nice.  What's involved with a CORS proxy?

**Comment by ylemkimon** [comment]
@kevinbarabash 
> Using build artifacts from Circle CI sounds nice. What's involved with a CORS proxy?

Sorry for late reply, I missed your comment. The build artifacts from CircleCI [doesn’t have CORS header set](https://discuss.circleci.com/t/add-cors-support-to-artifacts-server/806), i.e., allow loading from other domains. So we need a proxy like [cors-anywhere](https://cors-anywhere.herokuapp.com) or a third party object storage like AWS S3 to serve files with correct headers. However, I cannot find reliable proxy. So we need to setup a CORS proxy or an object storage. (CORS proxy with cache is essentially same as object storage)

**Comment by kevinbarabash** [comment]
@ylemkimon it seems like using `netlify` might be easier for hosting `master` docs, but we should probably secure a domain first.  I asked our IT dept about purchasing the domain name and they can totally do it but they need manager approval and my manager is on vacation.  @edemaine you mentioned being able to get the domain and transfer it later.  I'm not sure how that's done, but that's maybe the best option to move things forward.

**Comment by edemaine** [comment]
@kevinbarabash Preference on `katex.io` vs. `katexjs.org`?  See https://github.com/Khan/KaTeX/issues/1509#issuecomment-409772045 (and https://github.com/Khan/KaTeX/issues/1509#issuecomment-411278202 for an alternative).

**Comment by ylemkimon** [comment]
@edemaine I think the shorter, the better. If you and @kevinbarabash don't mind, I'd like to set up the domain to arrange with ongoing PRs.

**Pull request closed without merging**